examples/datasource-http-backend/pkg/plugin/datasource.go

```diff
@@ -0,0 +1,152 @@
+// be disposed and a new one will be created using NewSampleDatasource factory function.
+func (d *Datasource) Dispose() {
+	// Clean up datasource instance resources.
+	d.httpClient.CloseIdleConnections()
+}
+
+// QueryData handles multiple queries and returns multiple responses.
+// req contains the queries []DataQuery (where each query contains RefID as a unique identifier).
+// The QueryDataResponse contains a map of RefID to the response for each query, and each response
+// contains Frames ([]*Frame).
+func (d *Datasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	// create response struct
+	response := backend.NewQueryDataResponse()
+
+	// loop over queries and execute them individually.
+	for _, q := range req.Queries {
+		res, err := d.query(ctx, req.PluginContext, q)
+		if err != nil {
+			return nil, fmt.Errorf("query: %w", err)
+		}
+		// save the response in a hashmap
+		// based on with RefID as identifier
+		response.Responses[q.RefID] = res
+	}
+
+	return response, nil
+}
+
+func (d *Datasource) query(ctx context.Context, pCtx backend.PluginContext, query backend.DataQuery) (backend.DataResponse, error) {
+	// Response to be returned.
+	var response backend.DataResponse
+
+	// Do HTTP request
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, d.settings.URL, nil)
+	if err != nil {
+		return response, fmt.Errorf("new request with context: %w", err)
+	}
+	resp, err := d.httpClient.Do(req)
+	if err != nil {
+		return response, fmt.Errorf("http client do: %w", err)
+	}
+	defer resp.Body.Close()
+
+	// Make sure the response was successful
+	if resp.StatusCode != http.StatusOK {
+		return response, fmt.Errorf("expected 200 response, got %d", resp.StatusCode)
+	}
+
+	// Decode response
+	var body apiMetrics
+	if err := json.NewDecoder(resp.Body).Decode(&body); err != nil {
+		return response, fmt.Errorf("decode: %w", err)
+	}
+
+	// Create slice of values for time and values.
+	times := make([]time.Time, len(body.DataPoints))
+	values := make([]float64, len(body.DataPoints))
+	for i, p := range body.DataPoints {
+		times[i] = p.Time
+		values[i] = p.Value
+	}
+
+	// Create frame and add it to the response
+	response.Frames = append(
+		response.Frames,
+		data.NewFrame(
+			"response",
+			data.NewField("time", nil, times),
+			data.NewField("values", nil, values),
+		),
+	)
+	return response, nil
+}
+
+// CheckHealth performs a request to the specified data source and returns an error if the HTTP handler did not return
+// a 200 OK response.
+func (d *Datasource) CheckHealth(ctx context.Context, _ *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+	r, err := http.NewRequestWithContext(ctx, http.MethodGet, d.settings.URL, nil)
+	if err != nil {
+		return newHealthCheckErrorf("could not create request"), nil
+	}
+	resp, err := d.httpClient.Do(r)
+	if err != nil {
+		return newHealthCheckErrorf("request error"), nil
+	}
+	defer resp.Body.Close()
```

Comment:
Great. But would also like you to handle the case if `Body.Close` returns an error and log that. Otherwise those errors just being swallowed/ignored which is not great.